service/src/main/kotlin/no/nav/su/se/bakover/service/søknad/SøknadServiceImpl.kt

```diff
@@ -85,24 +99,82 @@ class SøknadServiceImpl(
                 ).also {
                     sakService.opprettSak(it)
                 }
-                val sakIdSaksnummerFnr = sakService.hentSakidOgSaksnummer(fnr)
+                val sakinfo = sakService.hentSakInfo(nySak.id)
                     .getOrElse { throw RuntimeException("Feil ved henting av sak") }
-                Pair(sakIdSaksnummerFnr, nySak.søknad)
-            },
-            {
+                Pair(sakinfo, nySak.søknad)
+            }
+            .map { saker ->
                 log.info("Ny søknad: Fant eksisterende sak for fødselsnummmer. Oppretter ny søknad på eksisterende sak.")
-                val søknad = Søknad.Ny(
-                    sakId = it.sakId,
-                    id = UUID.randomUUID(),
-                    opprettet = Tidspunkt.now(clock),
+                Either.catch { saker.single() }.fold(
+                    {
+                        val sak = saker.find { it.type == søknadInnhold.type() } ?: return KunneIkkeOppretteSøknad.FeilSakstype.left()
```

Comment:
Denne `KunneIkkeOppretteSøknad.FeilSakstype.left()` blir ikke representert i returverdien  for `Either.catch`